runtime/vm/romclasses.c

```diff
@@ -127,6 +127,10 @@ initializeArrayROMClass(J9ROMArrayClass *romClass, J9UTF8 *className, U_32 array
 	NNSRP_SET(romClass->className, className);
 	NNSRP_SET(romClass->superclassName, &arrayROMClasses.objectClassName);
 	romClass->modifiers = J9AccFinal | J9AccPublic | J9AccClassArray | J9AccAbstract;
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+	/* Arrays are always identity classes. */
+	romClass->modifiers |= J9AccClassHasIdentity;
```

Comment:
do we add the flag for non-Arrays as well?

Reply:
Yes, for non-array classes it is set here:

https://github.com/eclipse-openj9/openj9/blob/52ab41fa82505d2b245129429bd419591b7d17d1/runtime/bcutil/ClassFileOracle.cpp#L817-L819